### DISCUSSION TITLE
Trying to record a video, getting an error

### DISCUSSION THREAD

**Discussion by antongenis** [bluenviron/mediamtx#598] [Questions]
2021/09/29 11:26:18 I [1/0] [RTSP] [session 644619446] opened by 127.0.0.1:38766
[rtsp @ 0x5555c885b900] method SETUP failed: 461 Unsupported Transport**
2021/09/29 11:26:18 I [1/0] [RTSP] [session 158090036] opened by 127.0.0.1:38766

the command I am using to send a video:

ffmpeg -re -i ipd.mp4 -c:v libx264 -preset superfast -tune zerolatency -f rtsp -rtsp_transport tcp rtsp://SS:DD@52.233.254.0:8554/stream7    
___________________________________________________________________________________________
in the yml file:

runOnPublish: ffmpeg -i rtsp://localhost:$RTSP_PORT/$RTSP_PATH -rtsp_transport tcp -c:v libx264 /mnt/video-dev/$RTSP_PATH.mp4

**Comment by aler9** [comment]
Move the `-rtsp_transport` flag before `-i`:

```yml
runOnPublish: ffmpeg -rtsp_transport tcp -i rtsp://localhost:$RTSP_PORT/$RTSP_PATH -c:v libx264 /mnt/video-dev/$RTSP_PATH.mp4
```

[a comment by an automated account is not shown]
